client/src/client.rs

```diff
@@ -742,21 +744,30 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 
 		operation.op.insert_aux(aux)?;
 
+
+		let header = import_headers.into_post();
```

Comment:
Might be worth to construct this lazily for `all_blocks`. Also I'd consider passing less data to `all_blocks` notification for performance reasons.

Reply:
Passing less data is imho not possible since `header` is required in https://github.com/paritytech/substrate/pull/5455/files#diff-240c17a1e4dbce4ad137b7777bf03ce4R919-R924 which in turn requires `header` to be extracted from `import_headers` - correct me if I am wrong.

Reply:
In case there can be many *all-blocks-notification-streams* one could pass an `Arc<AllBlocksNotification>` down the channels instead of a clone for each channel within `notify_any_block_imported`.

We do similar things for the `NetworkService::event_stream` where we pass a `Bytes` instead of a `Vec<u8>`.